trio/tests/test_file_io.py

```diff
@@ -0,0 +1,169 @@
+import io
+import tempfile
+
+import pytest
+from unittest import mock
+from unittest.mock import patch, sentinel
```

Comment:
> APIs defined like that should be rewritten so that their guarantees are defined in terms of real things

I am no testing expert, but my thinking is that a unit test should test only the functionality of that unit; so the mocking and patching removes functionality that depends on other units.